pkgs/development/compilers/crystal/default.nix

```diff
@@ -328,5 +328,13 @@ rec {
     doCheck = false;
   };
 
+  crystal_1_17 = generic {
+    version = "1.17.1";
+    sha256 = "sha256-+wHhozPhpIsfQy1Lw+V48zvuWCfXzT4IC9KA1AU/DLw=";
+    binary = binaryCrystal_1_10;
+    llvmPackages = llvmPackages_18;
```

Comment:
FWIW, this supports up to LLVM 21 (though we haven’t bumped our LLVM 21 package to the stable release just yet): https://github.com/crystal-lang/crystal/blob/1.17.1/src/llvm/ext/llvm-versions.txt

(I have commits pending to bump other Crystal LLVM versions as part of dropping old LLVM versions, so just giving a heads up.)

Reply:
Thank you for the input. I chose LLVM 18 because I saw [an upstream commit](https://github.com/crystal-lang/crystal/commit/c2d1a017368bde7cd84db910a69a36cfd7cbfbae#diff-e53dfbfffe62ae3c0b411b3938ccffa9fb6a2ecc565f55785ef8daa756631a6bR22) that seemed to depend on older versions(11 -> 18 -> 11).

However, it would be great if a newer version works too. 👍